### PR TITLE
Rf crud

### PR DESCRIPTION
**Pull request by neymarjimoh**
**What does this PR do?**
- Contains functionalities for users to perform CRUD operations on their products.

**description of Task to be completed?**
- documentation of added features
- write test for the features
- add more features like search, get, delete etc

**How should this be manually tested?**
- after cloning this repo, run `npm install`
- checkout to this branch `git checkout rf-CRUD`
- run the development server `npm run dev`
- test the endpoints on postman `http://localhost/{port}/products/new` and `http://localhost/{port}/products/{productId}`

**Any background context you want to provide?**
None

**What are the relevant PT Story ID?**

[#173226660](https://www.pivotaltracker.com/story/show/173226660)

**Screenshots (if appropriate)**
product crea

**Questions:**
None

### CODE REVIEW

File: constants/routesGroup.js [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+        '/api/v1/auth/logout',
         '/api/v1/users/:userId',
+        '/api/v1/users/profile/update',
         '/api/v1/users/search/:name',
-    ]
+        '/api/v1/products/new',
+        '/api/v1/rofile/update/:id',
+        '/api/v1/users/email-subscription',
+    ],
 };
```
Comment (Nicanor008) [inline constants/routesGroup.js:27]: You should always leave a blank line at the end

File: controllers/user.controller.js [hunk 2]
```diff
@@ -4,13 +4,30 @@ const mongoose = require('mongoose');
...
+            users = await User.find({ role: 'INVESTOR'})
+                .populate('investments', 'product product.name')
+                .select('-password -passwordConfirmation');
+        } else if (role === 'INVESTOR') {
+            users = await User.find({ role: 'USER'})
+                .populate('investors', 'fullName email investorType')
+                .select('-password -passwordConfirmation');
+        } else {
+            //  ADMIN role
+            users = await User.find().sort({ createdAt: -1 });
```
Comment (Nicanor008) [inline controllers/user.controller.js:19]: Good implementation of this, you considered DRY principles

File: controllers/user.controller.js [hunk 3]
```diff
@@ -28,69 +45,90 @@ exports.getSingleUser = async (req, res) => {
...
-        const user = await User.findOneAndUpdate(
+
+        const profile = await Profile.findOneAndUpdate(
+            {
+                userId: id,
+            },
+            {
+                $set: {
+                    ...req.body,
+                    image: image,
```
Comment (Nicanor008) [inline controllers/user.controller.js:104]: you can use es7 syntax, you don't to redeclare the key and value of the object if it's the same
```suggestion
                    image,
```

**Comment by neymarjimoh** [comment]
Okay sir

File: validation/product.validation.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+// 		// .if((value, { req }) => req.body.newPassword)
+// 			// OR
+// 			.if(body('newPassword').exists())
+// 			// ...then the old password must be too...
+// 			.notEmpty()
+// 			// ...and they must not be equal.
+// 			.custom((value, { req }) => value !== req.body.newPassword),
+	
+// 	]
+// };
```
Comment (Nicanor008) [inline validation/product.validation.js:56]: Remove unused code or write a comment on top that it is to be re-used later

**Comment by neymarjimoh** [comment]
conflicts fixed @Nicanor008

**Review by Nicanor008 — APPROVED** [review]

**Pull request merged**